openfl/utilities/attestation/attestation_utils.py

```diff
@@ -0,0 +1,259 @@
+
+from base64 import b64encode
+import json
+import base64
+import os
+import http
+import requests
+import hashlib
+import logging
+
+from openfl.utilities.attestation.signer_utils import ECDSASigner
+
+
+logger = logging.getLogger(__name__)
+
+class AttestationManager:
+    """Class to manage attestation for enclaves.
+    This class handles the generation of SGX quotes, fetching MRENCLAVE values,
+    and obtaining attestation reports from AVS (Attestation Verification Service).
+    """
+    
+
+    def __init__(self, participant_name, attestation_report_path, ita_api_key, avs_url):
+        """Initializes the AttestationManager with the provided parameters.
+
+        Args:
+            participant_name (str): Name of the enclave.
+            attestation_report_path (str): Path to the store attestation reports.
+            ita_api_key (str): API key for ITA.
+            avs_url (str): URL for AVS.
+            enable_remote_attestation (bool): Flag to enable remote attestation.
+        """
+        self.participant_name = participant_name
+        self.attestation_report_path = attestation_report_path
+        self.ita_api_key = ita_api_key
+        self.avs_url = avs_url
+
+
+    def get_attestation_report(
+            self
+    ):
+        """Generates the attestation report for the enclave.
+        This function generates the SGX quote, fetches the MRENCLAVE value,
+        and obtains the attestation report from AVS.
+        It also generates an ECDSA-P 384 key pair and certificate for mTLS
+        with the collaborator enclave.
+        The generated report is saved in the specified attestation report path.
+        The function also checks if the required environment variables are set
+        and raises an exception if any of them are missing.
+        The function returns a dictionary containing the MRENCLAVE value,
+        participant name, and the AVS report token.
+        The function raises a FileNotFoundError if the attestation report path
+        does not exist, and a ValueError if the ITA API key or AVS URL is not set.
+        Args:
+            None
+        Raises:
+            FileNotFoundError: If the attestation report path does not exist.
+            ValueError: If the ITA API key or AVS URL is not set.
+        Returns:
+            dict: A dictionary containing the MRENCLAVE value, participant name,
+                  and AVS report token.
```

Comment:
PS: instead of a dict, I suggest returning an `AttestedIdentity` object here, which contains the following elements:
- name
- private key
- public key (cert)
- MRENCLAVE
- ITA report (token)

The `AttestedIdentity` object could also have a `get_passport()` method that would simply return everything but the private key (useful for saving the "attestation bundle" - effectively the enclave's passport).

Reply:
added a new class AttestedIdentity which can be used by openFl security as well